### ISSUE TITLE
Empty continuation lines

### ISSUE THREAD

**Issue by blueyed** [hadolint/hadolint#251]
Docker itself warns about empty continuation lines with docker-build.

Give `Dockerfile`:
```docker
FROM alpine:3.8
RUN true \

RUN false
```

```
[WARNING]: Empty continuation line found in:
    RUN true RUN false
[WARNING]: Empty continuation lines will become errors in a future release.
Step 1/2 : FROM alpine:3.8
 ---> 11cd0b38bc3c
Step 2/2 : RUN true RUN false
 ---> Running in dfb8a7f78ae9
Removing intermediate container dfb8a7f78ae9
 ---> c71baf6ad584
Successfully built c71baf6ad584
```

It would be nice if hadolint would trigger a warning/error for it also already.
Currently it does not trigger any issue for the above Dockerfile.

**Comment by lorenzo** [comment]
thanks, this is a good idea

**Comment by bdovaz** [comment]
@lorenzo any news? Thanks.

**Comment by Mohataseem89** [comment]
i was looking into this to understand what it would take to fix.
After reading the l`anguage-docker` parser source the reason this has not been fixed is that the information is lost at parse time, before any rule ever sees it The `escapedLineBreaks` parser in `Prelude.hs` uses `takeWhile1P Nothing isNl` to consume newlines during a line continuation it just silently swallows the blank line and moves on. by the time a rule runs, there's no trace that an empty continuation line ever existed

so unlike a typical rule-layer fix this would require:
1. Modifying `language-docker` to preserve "empty continuation" trivia in the AST
2. writing the actual hadolint rule to read those new AST nodes

That is a two-repo change across `hadolint/language-docker` and `hadolint/hadolint` which is probably why it's been sitting open. Flagging this so future contributors know the root cause upfront and don't spend time looking for a rule-only fix
happy to help if a maintainer wants to discuss the right approach for the parser change.
